cleanup-passes/travis.ts

```diff
@@ -32,56 +31,133 @@ async function cleanupTravisConfig(element: ElementRepo): Promise<void> {
 
   let travis = yaml.safeLoad(travisConfigBlob);
 
-  let needsReview = false;
+  // a small state class to represent the decision to make a PR or a commit to head
+  // Once tripped, a PR will be made
+  class Reviewer {
+    private _needsReview = false
+    get needsReview() {
+      return this._needsReview;
+    }
+    set needsReview(value: boolean) {
+      if (!this._needsReview) {
+        this._needsReview = value;
+      }
+    }
+  }
+
+  const review = new Reviewer();
+
+  const tools: string[] = [
+    'bower',
+    'polylint',
+    'web-component-tester'
+  ];
+
+  const beforeScript: string[] = [
+    `npm install -g ${tools.join(" ")}`,
+    'bower install',
+    'polylint'
+  ];
 
   // update travis config
   // Add polylint to all elements
   if (Array.isArray(travis.before_script)) {
-    const installStep = 'npm install polylint';
-    const runStep = 'polylint';
-
-    // assume we need a review
-    needsReview = true;
-
+    // assume we need review
     const bs: string[] = travis.before_script;
-    const installIndex = bs.indexOf(installStep);
-    const runIndex = bs.indexOf(runStep);
-
-    if (installIndex < 0 && runIndex < 0) {
-      // add both steps
-      bs.push(installStep);
-      bs.push(runStep);
-    } else if (runIndex >= 0 && installIndex < 0) {
-      // add install step before run step
-      bs.splice(runIndex, 0, installStep);
-    } else if (runIndex < 0 && installIndex >= 0) {
-      // add run step after install step
-      bs.push(runStep);
-    } else if (runIndex < installIndex) {
-      // reorder install step to be before run step
-      bs.splice(installIndex, 1);
-      bs.splice(runIndex, 0, installStep);
+    if (bs.length !== beforeScript.length) {
+      travis.before_script = beforeScript;
+      review.needsReview = true;
     } else {
-      // all is well
-      needsReview = false;
+      for (let i = 0; i < bs.length; i++) {
```

Comment:
Recommend moving this logic into a little deep equals function, then you can dedupe the before_script and needsReview setting